frontend/src/formdescriptioneditor/FormDescriptionEditorWebSocketContainer.tsx

```diff
@@ -0,0 +1,271 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { makeStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+import TextFieldsIcon from '@material-ui/icons/TextFields';
+import {
+  FormDescriptionEditorWebSocketContainerState,
+  Kind,
+  Widget,
+  WidgetEntryProps,
+  WidgetProps,
+} from 'formdescriptioneditor/FormDescriptionEditorWebSocketContainer.types';
+import React, { useState } from 'react';
+import { v4 as uuid } from 'uuid';
+import { RepresentationComponentProps, Selection } from 'workbench/Workbench.types';
+
+const useFormDescriptionEditorStyles = makeStyles((theme) => ({
+  formDescriptionEditor: {
+    display: 'flex',
+    flexDirection: 'column',
+
+    width: '100%',
+  },
+  header: {
+    padding: '4px 8px 4px 8px',
+  },
+  main: {
+    display: 'flex',
+    flexDirection: 'row',
+    height: '100%',
+    borderTop: '1px solid grey',
+  },
+  widgets: {
+    display: 'flex',
+    flexDirection: 'column',
+    borderRight: '1px solid grey',
+    padding: '4px 8px 4px 8px',
+  },
+  widgetKind: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    marginTop: '8px',
+    marginBottom: '8px',
+    cursor: 'move',
+  },
+  preview: {
+    width: '100%',
+    padding: '4px 8px 4px 8px',
+  },
+  body: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'stretch',
+  },
+  bottomDropArea: {
+    height: '100px',
+  },
+  dragOver: {
+    border: 'dashed 1px red',
+  },
+}));
+
+const isKind = (value: string): value is Kind => {
+  return value === 'Textfield';
+};
+
+export const FormDescriptionEditorWebSocketContainer = ({ selection, setSelection }: RepresentationComponentProps) => {
+  const classes = useFormDescriptionEditorStyles();
+  const initialState: FormDescriptionEditorWebSocketContainerState = {
+    widgets: [],
+  };
+  const [state, setState] = useState<FormDescriptionEditorWebSocketContainerState>(initialState);
```

Comment:
While it's not an issue for this first contribution, you will very quickly need to transition to a proper state machine here.

Reply:
Done

Reply:
I've made some minor adjustments to your code:

- I've aligned the naming convention of your events to other state machines (UpdateWidgetsEvent)
- I've removed the fact that widgets have children (this was some quick and dirt way to create containers which had child widgets)